src/bootchooser.c

```diff
@@ -614,6 +615,468 @@ static gboolean uboot_set_primary(RaucSlot *slot, GError **error) {
 	return res;
 }
 
+typedef struct {
+	gchar* num;
+	gchar* name;
+	gboolean active;
+} efi_bootentry;
+
+static gboolean efi_bootorder_set(gchar *order, GError **error) {
+	GSubprocess *sub;
+	GError *ierror = NULL;
+	gboolean res = FALSE;
+
+	g_return_val_if_fail(order, FALSE);
+	g_return_val_if_fail(error == NULL || *error == NULL, FALSE);
+
+
+	sub = g_subprocess_new(G_SUBPROCESS_FLAGS_NONE, &ierror, "efibootmgr",
+			       "--bootorder", order, NULL);
+
+	if (!sub) {
+		g_propagate_prefixed_error(
+				error,
+				ierror,
+				"Failed to start " EFIBOOTMGR_NAME ": ");
+		goto out;
+	}
+
+
+	res = g_subprocess_wait_check(sub, NULL, &ierror);
+	if (!res) {
+		g_propagate_prefixed_error(
+				error,
+				ierror,
+				"Failed to run " EFIBOOTMGR_NAME ": ");
+		goto out;
+	}
+
+out:
+	return res;
+}
+
+static gboolean efi_set_bootnext(gchar *bootnumber, GError **error) {
+	GSubprocess *sub;
+	GError *ierror = NULL;
+	gboolean res = FALSE;
+
+	g_return_val_if_fail(bootnumber, FALSE);
+	g_return_val_if_fail(error == NULL || *error == NULL, FALSE);
+
+	sub = g_subprocess_new(G_SUBPROCESS_FLAGS_NONE, &ierror, "efibootmgr",
+			       "--bootnext", bootnumber, NULL);
+
+	if (!sub) {
+		g_propagate_prefixed_error(
+				error,
+				ierror,
+				"Failed to start " EFIBOOTMGR_NAME ": ");
+		goto out;
+	}
+
+
+	res = g_subprocess_wait_check(sub, NULL, &ierror);
+	if (!res) {
+		g_propagate_prefixed_error(
+				error,
+				ierror,
+				"Failed to run " EFIBOOTMGR_NAME ": ");
+		goto out;
+	}
+
+out:
+	return res;
+}
+
+static efi_bootentry* get_efi_entry_by_bootnum(GList *entries, const gchar *bootnum) {
+	efi_bootentry *found_entry = NULL;
+
+	g_return_val_if_fail(entries, NULL);
+	g_return_val_if_fail(bootnum, NULL);
+
+	for (GList *entry = entries; entry != NULL; entry = entry->next) {
+		efi_bootentry *ptr = entry->data;
+		if (g_strcmp0(bootnum, ptr->num) == 0) {
+			found_entry = ptr;
+			break;
+		}
+	}
+
+	return found_entry;
+}
+
+static gboolean efi_bootorder_get(GList **bootorder_entries, GList **all_entries, efi_bootentry **bootnext, GError **error) {
```

Comment:
This needs a comment to explain the arguments, especially bootorder_entries and all_entries.